### PR TITLE
Unit tests for job->task-metadata

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

File: scheduler/src/cook/mesos/task.clj [hunk 1]
```diff
@@ -14,43 +14,41 @@
...
                "UTF-8")]
-    ;; If the there is no value for key :job/name, the following name will contain a substring "null".
     {:command command
      :container container
      :data data
      :environment environment
      :executor-key executor-key
      :framework-id fid
      :labels labels
+     ;; If the there is no value for key :job/name, the following name will contain a substring "null".
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/task.clj]: I don't understand this comment. Where is the null coming from?

File: scheduler/src/cook/mesos/task.clj [hunk 2]
```diff
@@ -14,43 +14,41 @@
...
-        command {:value (:job/command job-ent)
-                 :environment environment
+        command {:environment environment
+                 :uris (:uris resources [])
                  :user (:job/user job-ent)
-                 :uris (:uris resources [])}
+                 :value (:job/command job-ent)}
         ;; executor-key configure whether this is a command or custom executor
-        executor-key (if custom-executor :executor :command)
-        ;;TODO this data is a race-condition
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/task.clj]: The instance count is still a race condition

File: scheduler/src/cook/mesos/task.clj [hunk 3]
```diff
@@ -14,43 +14,41 @@
...
-  (:require [clojure.tools.logging :as log]
-            [cook.mesos.util :as util]
+  (:require [cook.mesos.util :as util]
             [plumbing.core :refer (map-vals)])
   (:import com.netflix.fenzo.TaskAssignmentResult))
 
-(defonce custom-executor-name "cook_agent_executor")
-(defonce custom-executor-source "cook_scheduler")
+(def custom-executor-name "cook_agent_executor")
+(def executor-source "cook_scheduler")
```
Comment (wyegelwel) [inline scheduler/src/cook/mesos/task.clj]: Why remove the defonce?

File: scheduler/src/cook/mesos/task.clj (hunk 1 above)
Comment (shamsimam) [inline scheduler/src/cook/mesos/task.clj]: I just moved this comment closer to the `:name` field from earlier in the file (line 46).

File: scheduler/src/cook/mesos/task.clj (hunk 2 above)
Comment (shamsimam) [inline scheduler/src/cook/mesos/task.clj]: Oops, added it back.

File: scheduler/src/cook/mesos/task.clj (hunk 3 above)
Comment (shamsimam) [inline scheduler/src/cook/mesos/task.clj]: I didn't find these being defined elsewhere. Are they were not intentionally set to `defonce`?

I'll add the `defonce` back.

File: scheduler/src/cook/mesos/task.clj (hunk 1 above)
Comment (shamsimam) [inline scheduler/src/cook/mesos/task.clj]: Removed it altogether as `:job/name` is looked up with a default value.

**Comment by shamsimam** [comment]
@wyegelwel addressed your feedback

File: scheduler/test/cook/test/mesos/task.clj [hunk 4]
```diff
@@ -115,98 +119,323 @@
...
+                                    :mode "RW"}]
+                         :docker {:image "nvidia/cuda"
+                                  :network "HOST"
+                                  :force-pull-image false
+                                  :parameters [{:key "user" :value "100:5"}]
+                                  :port-mapping [{:host-port 0
+                                                  :container-port 1
+                                                  :protocol "tcp"}]}}
+              container-task (assoc task :container container)
+              ;; TODO: Check values of container-msg.
```
Comment (pschorf) [inline scheduler/test/cook/test/mesos/task.clj]: Can this be done? :)

Comment (shamsimam) [inline scheduler/test/cook/test/mesos/task.clj]: Done :)

**Comment by wyegelwel** [comment]
@shamsimam happy to merge if there is nothing else you intend to change.

**Comment by shamsimam** [comment]
@wyegelwel please merge :)

**Pull request merged**